src/main/java/com/conveyal/r5/analyst/Grid.java

```diff
@@ -126,96 +126,154 @@ public Grid (int zoom, int width, int height, int north, int west) {
         this.grid = new double[width][height];
     }
 
+    public static class PixelWeight {
+        int x;
+        int y;
+        double weight;
+
+        private PixelWeight (int x, int y, double weight){
+            this.x = x;
+            this.y = y;
+            this.weight = weight;
+        }
+
+        public int getX() {
+            return x;
```

Comment:
I would favor final fields, set in the constructor, without accessor methods ("getters"). This is the approach we've been using in most other Java code that doesn't need serious encapsulation of fields.